src/connectionconfig/connectionDialogWebviewController.ts

```diff
@@ -822,21 +823,40 @@ export class ConnectionDialogWebviewController extends FormWebviewController<
                 cleanedConnection as any,
             );
 
-            await this._mainController.connectionManager.connectionStore.saveProfile(
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                cleanedConnection as any,
-            );
-            const node = await this._mainController.createObjectExplorerSession(cleanedConnection);
-            await this.updateLoadedConnections(state);
-            this.updateState();
+            async function saveConnectionAndCreateSession(
+                self: ConnectionDialogWebviewController,
+            ): Promise<TreeNodeInfo> {
+                await self._mainController.connectionManager.connectionStore.saveProfile(
+                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                    cleanedConnection as any,
+                );
+                const node =
+                    await self._mainController.createObjectExplorerSession(cleanedConnection);
+                await self.updateLoadedConnections(state);
+                self.updateState();
+
+                return node;
+            }
+
+            let node = await saveConnectionAndCreateSession(this);
 
             this.state.connectionStatus = ApiStatus.Loaded;
 
-            await this._mainController.objectExplorerTree.reveal(node, {
-                focus: true,
-                select: true,
-                expand: true,
-            });
+            try {
```

Comment:
Should we add an event in the OE provider that is fired whenever we refresh a tree item. We can listen to the event and only continue when we refresh for the right parent is done. 

I also like your idea of making config writes from the extension unidirectional.

Reply:
ObjectExplorerService already has `_onDidChangeTreeData(node)` that triggers upon refresh, but I think leveraging that for this particular scenario might result in more event spaghetti.  My hope is that driving the config write eventing from the bottom-up would untangle much of that.